File: tool/uapicodegen/uapicodegen/uapi/internal/mock/IsSubMapEntryEqual.py
def is_sub_map_entry_equal(part_value: object, whole_value: object) -> bool:
    from ...internal.mock.IsSubMap import is_sub_map
    from ...internal.mock.PartiallyMatches import partially_matches

    if isinstance(part_value, dict) and isinstance(whole_value, dict):
        return is_sub_map(part_value, whole_value)
    elif isinstance(part_value, list) and isinstance(whole_value, list):
        for part_element in part_value:
            part_matches = partially_matches(whole_value, part_element)
            if not part_matches:
                return False
        return True
    else:
        return part_value == whole_value

File: lib/java/src/main/java/io/github/telepact/internal/mock/IsSubMapEntryEqual.java
package io.github.telepact.internal.mock;

import static io.github.telepact.internal.mock.IsSubMap.isSubMap;
import static io.github.telepact.internal.mock.PartiallyMatches.partiallyMatches;

import java.util.List;
import java.util.Map;
import java.util.Objects;

public class IsSubMapEntryEqual {
    public static boolean isSubMapEntryEqual(Object partValue, Object wholeValue) {
        if (partValue instanceof final Map m1 && wholeValue instanceof final Map m2) {
            return isSubMap(m1, m2);
        } else if (partValue instanceof final List partList && wholeValue instanceof final List wholeList) {
            for (int i = 0; i < partList.size(); i += 1) {
                final var partElement = partList.get(i);
                final var partMatches = partiallyMatches(wholeList, partElement);
                if (!partMatches) {
                    return false;
                }
            }

            return true;
        } else {
            return Objects.equals(partValue, wholeValue);
        }
    }
}
